### PR TITLE
Qr code display migration

### PR DESCRIPTION
**Pull request by AvneetSingh2001**
Jira Task: [MM-18](https://mifosforge.jira.com/browse/MM-18)

| XML Screen (before)                                                                                               |  Compose Screen (Now)                                                                                            | 
|-------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------|
| Screenshot_20240615_033948 | Screenshot_20240615_033831 |

[MM-18]: https://mifosforge.jira.com/browse/MM-18?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/org/mifos/mobile/ui/qr_code_display/QrCodeDisplayTopBar.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.res.stringResource
+import org.mifos.mobile.R
+import org.mifos.mobile.core.ui.component.MifosTopBar
+
+@Composable
+fun QrCodeDisplayTopBar(
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobile/ui/qr_code_display/QrCodeDisplayTopBar.kt]: This should be generic type, Current generic type topbar in designsystem can be used here

File: app/src/main/java/org/mifos/mobile/ui/qr_code_display/QrCodeDisplayViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            flow {
+                val qrBitmap = QrCodeGenerator.encodeAsBitmap(str = qrString)
+                if(qrBitmap == null) {
+                    emit(QrCodeDisplayUiState.Error())
+                } else {
+                    emit(QrCodeDisplayUiState.Success(qrBitmap = qrBitmap))
+                }
+            }
+        }.catch {
+            emit(QrCodeDisplayUiState.Error(errorMessage = it.message))
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobile/ui/qr_code_display/QrCodeDisplayViewModel.kt:45]: Let's add a module in core:logs and define a method to register firebase non fatel error. On error send firebase crash error so we can know that this is working for everyone.

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/qr_code_display/QrCodeDisplayViewModel.kt (hunk 2 above)
Comment (AvneetSingh2001) [reply app/src/main/java/org/mifos/mobile/ui/qr_code_display/QrCodeDisplayViewModel.kt:45]: updated

File: app/src/main/java/org/mifos/mobile/ui/qr_code_display/QrCodeDisplayTopBar.kt (hunk 1 above)
Comment (AvneetSingh2001) [reply app/src/main/java/org/mifos/mobile/ui/qr_code_display/QrCodeDisplayTopBar.kt]: updated

**Pull request merged**